superset/security.py

```diff
@@ -92,6 +99,42 @@ def can_access(self, permission_name, view_name, user=None):
             return self.is_item_public(permission_name, view_name)
         return self._has_view_access(user, permission_name, view_name)
 
+    def has_method_access(self, f):
```

Comment:
It might just be me but I don't like the idea of a decorator as a class method. It's probably ok though. I guess decorators are already complex in terms of what's inside them, external context make it more of a stretch.